src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -1704,12 +1713,15 @@ private final class MapEntryIterator implements Iterator<MapEntry> {
         private MapEntry nextSpecial;
 
         private boolean vanityPathPrecedence;
+        private final Function<String, List<MapEntry>> getCurrentMapEntryForVanityPath;
 
-        public MapEntryIterator(final String startKey, final Map<String, List<MapEntry>> resolveMapsMap, final boolean vanityPathPrecedence) {
+        public MapEntryIterator(final String startKey, @NotNull final List<MapEntry> globalList,
+                                final Function<String, List<MapEntry>> getCurrentMapEntryForVanityPath,
```

Comment:
Right now I don't see the need for the parameter ``getCurrentMapEntryForVanityPath``, as this method is only used in this class. 
Will this change in the future?